spec/recurly/client_spec.rb

```diff
@@ -272,50 +272,85 @@
+              "message": "Something happened"
+            }
           }
-        }
-        JSON
+          JSON
+        end
+        allow(resp).to receive(:content_type).and_return "application/json; charset=utf-8"
+        resp_headers.each { |key, v| resp[key] = v }
+        resp
       end
-      allow(resp).to receive(:content_type).and_return "application/json; charset=utf-8"
-      resp_headers.each { |key, v| resp[key] = v }
-      resp
-    end
 
-    describe "#get" do
-      it "should raise an APIError" do
-        expect(net_http).to receive(:request).twice.and_return(response)
-        expect {
-          subject.get_account(account_id: "code-benjamin-du-monde")
-        }.to raise_error(Recurly::Errors::InternalServerError)
+      describe "#get" do
+        it "should raise an APIError" do
+          expect(net_http).to receive(:request).twice.and_return(response)
+          expect {
+            subject.get_account(account_id: "code-benjamin-du-monde")
+          }.to raise_error(Recurly::Errors::InternalServerError)
+        end
       end
-    end
 
-    context "retries" do
-      let(:successful_response) do
-        resp = Net::HTTPOK.new(1.0, "200", "OK")
-        allow(resp).to receive(:body) do
-          "{ \"object\": \"account\" }"
+      context "retries" do
+        let(:successful_response) do
+          resp = Net::HTTPOK.new(1.0, "200", "OK")
+          allow(resp).to receive(:body) do
+            "{ \"object\": \"account\" }"
+          end
+          allow(resp).to receive(:content_type).and_return "application/json; charset=utf-8"
+          resp_headers.each { |key, v| resp[key] = v }
+          resp
+        end
+
+        it "should retry and return success" do
+          expect(net_http).to receive(:request).and_return(response)
+          expect(net_http).to receive(:request).and_return(successful_response)
+          account = subject.get_account(account_id: "code-benjamin du monde")
+          expect(account).to be_instance_of Recurly::Resources::Account
         end
-        allow(resp).to receive(:content_type).and_return "application/json; charset=utf-8"
-        resp_headers.each { |key, v| resp[key] = v }
-        resp
       end
+    end
 
-      it "should retry and return success" do
-        expect(net_http).to receive(:request).and_return(response)
-        expect(net_http).to receive(:request).and_return(successful_response)
-        account = subject.get_account(account_id: "code-benjamin du monde")
-        expect(account).to be_instance_of Recurly::Resources::Account
+    context "unknown errors" do
+      let(:account_id) { "1234" }
+      let(:invoice_preview_path) { "/accounts/#{account_id}/invoices/preview" }
+      describe "#post" do
+        let(:response) do
+          resp = Net::HTTPConflict.new(1.0, "409", "Conflict")
+          allow(resp).to receive(:body) do
+            <<-JSON
+            {
+              "error": {
+                "type": "service_not_available",
```

Comment:
Thanks for putting this together, @ruyrocha. We are going to add `service_not_available` to the list of known error types, but think that your change is still beneficial for future purposes. Could you update this to be:
```json
"error": {
  "type": "unknown_error_type_code",
   "message": "An Unknow Error type code has been returned"
}
```

Reply:
@douglasmiller aside from the error message change, is there anything different for the HTTP error status code? I used 409 as there wasn't anything clear on headers, just the error message

Reply:
Yeah, let's change the status code to `"0"` in case we later add handling in `from_response` for a `409`. The next major version of the client will be using the generated [`ERROR_MAP`](https://github.com/recurly/recurly-client-ruby/blob/master/lib/recurly/errors/api_errors.rb#L7) and a `0` will be future proof.

Thanks again!

Reply:
Thanks, @douglasmiller!!! It's fixed on 65f9bf3